### ISSUE TITLE
Reports placed on wrong file

### ISSUE THREAD

**Issue by simon-katz** [clj-kondo/clj-kondo#1366]
**version**

v2021.08.06

**macro usage**

N/A. Not related to a macro of mine.

**platform**

macOS Catalina 10.15.7.
Native

**editor**

N/A. I get this behaviour at the command line.

**problem**

Some problems are reported as being in the wrong file. For details see the README in the repo in the repro section below.

**repro**

https://github.com/simon-katz/bug-report-clj-kondo-misplaced-reports

**config**

No config needed.

**expected behavior**

I expect to see all problems reported as being in the file `dir-1/user.clj`.

**Comment by borkdude** [comment]
Note to self:

When first linting `dir-2` and then `dir-1` the results seem to be more correct:

```
$ clj-kondo --lint dir-2 dir-1
dir-1/user.clj:3:24: warning: use alias or :refer
dir-1/user.clj:4:5: warning: namespace clojure.string is required but never used
dir-1/user.clj:4:28: warning: #'clojure.string/blank? is referred but never used
```

**Comment by borkdude** [comment]
Should be fixed on master.
